### ISSUE TITLE
Create Interactive Coding Demo

### ISSUE THREAD

**Issue by Palashio** [Palashio/awesome-libra#93]
<open-swe-issue-content>can we create an interactive coding demo</open-swe-issue-content>
<details>
<summary>Agent Context</summary>

<open-swe-do-not-edit-task-plan>
{
  "tasks": [
    {
      "id": "d8686fc9-068c-4820-a012-5268a2f04344",
      "taskIndex": 0,
      "request": "[original issue]\n**Create Interactive Coding Demo**\ncan we create an interactive coding demo",
      "title": "Create an interactive coding demo for the Libra machine learning library",
      "createdAt": 1753383477858,
      "completed": false,
      "planRevisions": [
        {
          "revisionIndex": 0,
          "plans": [
            {
              "index": 0,
              "plan": "Create an interactive HTML demo page (demo.html) that showcases Libra's one-line machine learning capabilities with embedded code examples, explanations for different user skill levels (beginners, experienced developers, ML engineers), and interactive elements like collapsible sections for different complexity levels",
              "completed": true,
              "summary": "Successfully created an interactive HTML demo page (demo.html) that showcases Libra's one-line machine learning capabilities. The demo includes:\n\n1. **Comprehensive Structure**: Created a fully responsive HTML page with modern CSS styling, gradient backgrounds, and smooth animations\n2. **Multi-level Content**: Organized content for three skill levels:\n   - Beginner: Simple one-line examples with explanations\n   - Intermediate: Custom parameters and configuration options\n   - Advanced: Full customization with hyperparameter tuning and neural networks\n3. **Interactive Elements**: Implemented collapsible sections using JavaScript for different complexity levels, allowing users to expand/collapse detailed explanations\n4. **Code Examples**: Embedded realistic Python code blocks showing:\n   - Basic regression: `client('housing_data.csv').regression('price')`\n   - Classification with parameters\n   - Neural network configurations\n   - Advanced ML pipelines with custom parameters\n5. **Visual Design**: Applied professional styling with:\n   - Color-coded skill level badges (green for beginner, orange for intermediate, red for advanced)\n   - Syntax-highlighted code blocks\n   - Feature cards in a responsive grid layout\n   - Hover effects and smooth transitions\n6. **Educational Content**: Included explanations of what happens behind the scenes, installation instructions, and next steps for users\n7. **Navigation**: Added smooth scrolling and interactive buttons linking to documentation and other demo files\n\nThe HTML file is 17KB and provides a complete interactive experience that demonstrates Libra's capabilities across all user skill levels while maintaining an engaging and professional presentation."
            },
            {
              "index": 1,
              "plan": "Create a comprehensive Python demo script (demo.py) that demonstrates Libra's key features including data preprocessing, model training, and evaluation with sample datasets, complete with comments explaining each step and different usage patterns for various skill levels",
              "completed": false
            },
            {
              "index": 2,
              "plan": "Create a Jupyter notebook demo (interactive_demo.ipynb) that provides hands-on examples of Libra usage with sample data, step-by-step explanations, and interactive cells that users can modify and run to experiment with different parameters",
              "completed": false
            },
            {
              "index": 3,
              "plan": "Update the README.md file to add a new 'Interactive Demo' section that links to the created demo files and provides instructions on how to run them locally",
              "completed": false
            }
          ],
          "createdAt": 1753383477858,
          "createdBy": "agent"
        }
      ],
      "activeRevisionIndex": 0
    }
  ],
  "activeTaskIndex": 0
}
</open-swe-do-not-edit-task-plan>

<open-swe-do-not-edit-proposed-plan>
[
  "Create an interactive HTML demo page (demo.html) that showcases Libra's one-line machine learning capabilities with embedded code examples, explanations for different user skill levels (beginners, experienced developers, ML engineers), and interactive elements like collapsible sections for different complexity levels",
  "Create a comprehensive Python demo script (demo.py) that demonstrates Libra's key features including data preprocessing, model training, and evaluation with sample datasets, complete with comments explaining each step and different usage patterns for various skill levels",
  "Create a Jupyter notebook demo (interactive_demo.ipynb) that provides hands-on examples of Libra usage with sample data, step-by-step explanations, and interactive cells that users can modify and run to experiment with different parameters",
  "Update the README.md file to add a new 'Interactive Demo' section that links to the created demo files and provides instructions on how to run them locally"
]
</open-swe-do-not-edit-proposed-plan>

</details>
</details>
